cmd/mattermost/commands/config.go

```diff
@@ -199,37 +210,34 @@ func printConfigValues(configMap map[string]interface{}, configSetting []string,
 }
 
 func configSetCmdF(command *cobra.Command, args []string) error {
-	app, err := InitDBCommandContextCobra(command)
+	configStore, err := getConfigStore(command)
 	if err != nil {
 		return err
 	}
 
-	defer app.Shutdown()
-
 	// args[0] -> holds the config setting that we want to change
 	// args[1:] -> the new value of the config setting
 	configSetting := args[0]
 	newVal := args[1:]
 
 	// create the function to update config
-	oldConfig := app.Config()
-	newConfig := app.Config()
-	f := updateConfigValue(configSetting, newVal, oldConfig, newConfig)
+	oldConfig := configStore.Get()
+	newConfig := configStore.Get()
 
-	app.UpdateConfig(f)
-	if err := newConfig.IsValid(); err != nil {
-		return err
+	f := updateConfigValue(configSetting, newVal, oldConfig, newConfig)
```

Comment:
This machinery could probably be simplified now that we're no longer dependent on `app.UpdateConfig`, but I left it in tact for now until we explicitly support `Patch` in the config store.